### ISSUE TITLE
False positive for `Lint/RedundantSafeNavigation` with `unless`

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#15147]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/RedundantSafeNavigation

### Steps to reproduce the problem

example.rb:
```ruby
a = nil

unless a&.ready?
  p a&.name
end
```

run `bundle exec rubocop example.rb --only Lint`. 

The `a&.name` is falsely reported. 

I believe the culprit is that `NilReceiverChecker#sole_condition_of_parent_if?` doesn't distinguish between `if` and `unless` (if the same code would have `if a&.ready?`, the deduction "if we are inside the conditional, it can't be `nil`" would be correct).

### Expected behavior

There should be no Lint-class problems with this code.

### Actual behavior

```
example.rb:4:6: W: [Correctable] Lint/RedundantSafeNavigation: Redundant safe navigation on non-nil receiver (detected by analyzing previous code/method invocations).
  p a&.name
     ^^
```

Autocorrection makes the code incorrect:
```ruby
a = nil

unless a&.ready?
  p a.name
end
```
Before: `ruby example.rb` would output `nil` (as it should).

After the autocorrection, it will be 
```
example.rb:4:in `<main>': undefined method `name' for nil (NoMethodError)

  p a.name
     ^^^^^
```

### RuboCop configuration

```yaml
Lint/RedundantSafeNavigation:
  InferNonNilReceiver: true
```

### RuboCop version

```shell
1.86.1 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 3.1, running on ruby 3.3.7) [x86_64-linux]

(Actually running from the latest `master`)
```
